web/pages/Chat/ChatDetail.tsx

```diff
@@ -57,17 +51,29 @@ const ChatDetail: Component = () => {
 
   const clickSwipe = (dir: -1 | 1) => () => {
     const ret = retries()
-    if (!ret || !ret.list.length) return
+
     const prev = swipe()
-    const max = ret.list.length - 1
+    const max = ret!.list.length
 
     let next = prev + dir
-    if (next < 0) next = max
-    else if (next > max) next = 0
+    
+    if (next <= 0) next = max
+    else if (next > max) msgStore.retry(id)
 
     setSwipe(next)
   }
 
+  window.onkeydown = (e: KeyboardEvent): any => {
```

Comment:
This behaviour will need to be done in `onMount` then removed when navigating away using `onCleanup`